src/client.ts

```diff
@@ -6772,6 +6775,29 @@ export class MatrixClient extends EventEmitter {
         return this.http.request(callback, Method.Post, "/register", params, data);
     }
 
+    /**
+     * Refreshes an access token using a provided refresh token. The refresh token
+     * must be valid for the current access token known to the client instance.
+     *
+     * Note that this function will not cause a logout if the token is deemed
+     * unknown by the server - the caller is responsible for managing logout
+     * actions on error.
+     * @param {string} refreshToken The refresh token.
+     * @return {Promise<IRefreshTokenResponse>} Resolves to the new token.
+     * @return {module:http-api.MatrixError} Rejects with an error response.
+     */
+    public refreshToken(refreshToken: string): Promise<IRefreshTokenResponse> {
+        return this.http.authedRequest(
+            undefined, Method.Post, "/_matrix/client/v1/refresh",
+            undefined, {
+                refresh_token: refreshToken,
+            }, {
+                prefix: "", // set manually
+                inhibitLogoutEmit: true, // we don't want to cause logout loops
+            },
+        );
```

Comment:
can we be consistent with how many args we do per line?

Reply:
can I say no? :P 

(I don't know what I was doing here tbh)